### ISSUE TITLE
find_matches() missing 2 required positional arguments: 'requirements' and 'incompatibilities'

### ISSUE THREAD

**Issue by sumitsharansatsangi** [pdm-project/pdm#598]
## Steps to reproduce
pdm add pkg_name

tried with a lots of package, but received the same error,
```
Adding packages to default dependencies: numpy
See /tmp/pdm-lock-7rby4o_t.log for detailed debug log.
[TypeError]: find_matches() missing 2 required positional arguments: 'requirements' and 'incompatibilities'
Add '-v' to see the detailed traceback
```

## Actual behavior
This is the actual behaviour
```
┌─[sumit@kumpali]─[~/fastapi_project/graphapp]
└──╼ $pdm add graphviz -v
Adding packages to default dependencies: graphviz
======== Start resolving requirements ========
  graphviz
  Adding requirement graphviz
Traceback (most recent call last):
  File "/usr/share/python-wheels/resolvelib-0.5.4-py2.py3-none-any.whl/resolvelib/resolvers.py", line 171, in _merge_into_criterion
KeyError: 'graphviz'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sumit/.local/bin/pdm", line 8, in <module>
    sys.exit(main())
  File "/home/sumit/.local/share/pdm/venv/lib/python3.9/site-packages/pdm/core.py", line 194, in main
    return Core().main(args)
  File "/home/sumit/.local/share/pdm/venv/lib/python3.9/site-packages/pdm/core.py", line 152, in main
    raise cast(Exception, err).with_traceback(traceback)
  File "/home/sumit/.local/share/pdm/venv/lib/python3.9/site-packages/pdm/core.py", line 147, in main
    f(options.project, options)
  File "/home/sumit/.local/share/pdm/venv/lib/python3.9/site-packages/pdm/cli/commands/add.py", line 51, in handle
    actions.do_add(
  File "/home/sumit/.local/share/pdm/venv/lib/python3.9/site-packages/pdm/cli/actions.py", line 208, in do_add
    resolved = do_lock(project, strategy, tracked_names, reqs)
  File "/home/sumit/.local/share/pdm/venv/lib/python3.9/site-packages/pdm/cli/actions.py", line 68, in do_lock
    mapping, dependencies, summaries = resolve(
  File "/home/sumit/.local/share/pdm/venv/lib/python3.9/site-packages/pdm/resolver/core.py", line 30, in resolve
    result = resolver.resolve(requirements, max_rounds)
  File "/usr/share/python-wheels/resolvelib-0.5.4-py2.py3-none-any.whl/resolvelib/resolvers.py", line 453, in resolve
  File "/usr/share/python-wheels/resolvelib-0.5.4-py2.py3-none-any.whl/resolvelib/resolvers.py", line 318, in resolve
  File "/usr/share/python-wheels/resolvelib-0.5.4-py2.py3-none-any.whl/resolvelib/resolvers.py", line 173, in _merge_into_criterion
  File "/usr/share/python-wheels/resolvelib-0.5.4-py2.py3-none-any.whl/resolvelib/resolvers.py", line 79, in from_requirement
TypeError: find_matches() missing 2 required positional arguments: 'requirements' and 'incompatibilities'
```

## Expected behavior

It was expected to add or download the package , but it generates the mentioned error.

## Environment Information
```
`Output of pdm info`
PDM version:        1.7.2                                                  
Python Interpreter: /usr/bin/python3 (3.9)                                 
Project Root:       /home/sumit/fastapi_project/graphapp                   
Project Packages:   /home/sumit/fastapi_project/graphapp/__pypackages__/3.9
```

```
`Output of pdm info --env`
{
  "implementation_name": "cpython",
  "implementation_version": "3.9.2",
  "os_name": "posix",
  "platform_machine": "x86_64",
  "platform_release": "5.10.0-8parrot1-amd64",
  "platform_system": "Linux",
  "platform_version": "#1 SMP Debian 5.10.46-8parrot1 (2021-07-21)",
  "python_full_version": "3.9.2",
  "platform_python_implementation": "CPython",
  "python_version": "3.9",
  "sys_platform": "linux"
}
```

**Comment by sumitsharansatsangi** [comment]
This error occurs when I have installed using the following command.
```
curl -sSL https://raw.githubusercontent.com/pdm-project/pdm/main/install-pdm.py | python -
```
But , when I tried with `pip install pdm` works.

I am badly surprised that the error occurs in the recommended way, One reason may be the parrot system, because manages the things in their own way, not sure.
Anyway, it works.
I want to know where the installed packages will be stored (consider reference as debian system), because sometimes, I need to change in the source file of the installed package to it in my custom way.  

Anyway, it works
